interface/src/app/tribe_client/genesets.js

```diff
@@ -0,0 +1,88 @@
+angular.module('adage.tribe_client.genesets', [
+  'adage.tribe_client.resource'
+])
+
+// Geneset Search Factory
+// Allows for retreiving for and interacting with search results for genesets.
+.factory('GenesetSearch', function($rootScope, Genesets) {
+  var genesets = [];
+  var query = {};
+  var totalResults = 0;
+
+  return {
+    getQuery: function() {
+      return query;
+    },
+    getGenesets: function() {
+      return genesets;
+    },
+    clear: function() {
+      query = {};
+      genesets = [];
+      $rootScope.$broadcast('genesets.update');
```

Comment:
It should be possible to have Angular do this cross-scope communication for you if you bind `genesets` to a relevant `$scope` in the DOM. Anyone who is currently configured to "receive" the `$broadcast` will automatically update via two-way binding.

This means a change to the `scope` parameter in the relevant `.directive`. The parent view that invokes the directive will also need to be updated to pass in the variable. I think of this like parameters to a function, only the "function" happens to be a visible widget. A place where I have implemented something like this is the `results` parameter in `interface/src/app/analyze/search/search.js`, with invoking parent in `.../analyze/analyze.tpl.html`. See if that example would fit this code.